File: LinearRegression_Salary.py
# -*- coding: utf-8 -*-
"""
Created on Sun Aug 18 00:28:13 2019

@author: Sumeet Kumar
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv("Salary_Data.csv")
X = dataset.iloc[:, :-1].values
Y = dataset.iloc[:, 1].values


from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.33,
                                                    random_state = 0)



# Fitting simple regression model 
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, Y_train)


#Predict the test set result
y_pred = regressor.predict(X_test)

#Visualize the result for train
plt.scatter(X_train, Y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary Vs Experience (Training Set)')
plt.xlabel('Year of Experience')
plt.ylabel('Salary')
plt.show()

#Visualize the result for test
plt.scatter(X_test, Y_test, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary Vs Experience (Training Set)')
plt.xlabel('Year of Experience')
plt.ylabel('Salary')
plt.show()

File: LinearRegression_Salary.R
# Sumeet Kumar
library(caTools)
library(ggplot2)


setwd("D:/Machine Learning/P14-Machine-Learning-AZ-Template-Folder/Machine Learning A-Z Template Folder/Part 2 - Regression/Section 4 - Simple Linear Regression/")


dataset = read.csv('Salary_Data.csv')

#dataset = dataset[, 2:3]


#train test split

set.seed(123)

split = sample.split(dataset$Salary, SplitRatio = 2/3)

training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#fit Linear regression line
regressor = lm(formula = Salary ~ YearsExperience, data = training_set)

#Prediction
y_pred = predict(regressor, newdata = test_set)

#visualize the graph for training set
ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             color = 'red') +
  
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), 
            color = 'blue') +
  ggtitle('Salary Vs Experience (Training Set)') + 
  xlab('Years of Experience') + 
  ylab('Salary')


#visualize the graph for test set
ggplot() + 
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             color = 'red') +
  
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), 
            color = 'blue') +
  ggtitle('Salary Vs Experience (Test Set)') + 
  xlab('Years of Experience') + 
  ylab('Salary')








